src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,399 @@
+                    }
+                    else if (parameters[parameterIndex].IsParams && parameters.Last().Type is IArrayTypeSymbol paramsType &&
+                      semanticModel.Compilation.ClassifyCommonConversion(argType.Type, paramsType.ElementType).Exists)
+                    {
+                        // For the parameter with keyword params, compare its element type.
+                        matchedTypes[parameterIndex] = true;
+                    }
+                    else return false;
+
+                    if (targetArgument.Equals(arguments[i])) targetParamIndex = parameterIndex;
+                }
+                else return false;
+            }
+
+            // mark all optional parameters as matched
+            for (var i = 0; i < parameters.Length; i++)
+            {
+                if (parameters[i].IsOptional || parameters[i].IsParams)
+                {
+                    matchedTypes[i] = true;
+                }
+            }
+
+            return Array.TrueForAll(matchedTypes, (item => item));
+        }
+
+        class InheritanceDistanceComparer : IComparer<ITypeSymbol>
+        {
+            private ITypeSymbol baseType;
+            private SemanticModel semanticModel;
+
+            private int GetInheritanceDistance(ITypeSymbol baseType, ITypeSymbol? derivedType)
+            {
+                if (derivedType == null) return int.MaxValue;
+                if (derivedType.Equals(baseType)) return 0;
+
+                var distance = int.MaxValue;
+                distance = Math.Min(GetInheritanceDistance(baseType, derivedType.BaseType), distance);
+
+                if (derivedType.Interfaces.Length != 0)
+                {
+                    foreach (var interface_type in derivedType.Interfaces)
+                    {
+                        distance = Math.Min(GetInheritanceDistance(baseType, interface_type), distance);
+                    }
+                }
+
+                return distance == int.MaxValue ? distance : distance + 1;
+            }
+            public int Compare(ITypeSymbol x, ITypeSymbol y)
+            {
+                // if the node has the explicit conversion operator, then it has the shortest distance
+                var x_dist = semanticModel.Compilation.ClassifyCommonConversion(baseType, x).IsUserDefined ?
+                    0 : GetInheritanceDistance(baseType, x);
+                var y_dist = semanticModel.Compilation.ClassifyCommonConversion(baseType, y).IsUserDefined ?
+                    0 : GetInheritanceDistance(baseType, y);
+                return x_dist.CompareTo(y_dist);
+            }
+
+            public InheritanceDistanceComparer(SemanticModel semanticModel, ITypeSymbol baseType)
+            {
+                this.semanticModel = semanticModel;
+                this.baseType = baseType;
+            }
+        }
+
+
+        protected SyntaxNode? TryGetTargetNode(SyntaxNode root, TextSpan span)
```

Comment:
Can you just inline this method into the caller: `if (root.FindNode(span)?.GetAncestors<ExpressionSyntax>().FirstOrDefault() is ExpressionSyntax targetNode)`